packages/programs/src/programs/DefaultProgram.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { IBaseWorld, WorldConsumer } from "@latticexyz/world-consumer/src/experimental/WorldConsumer.sol";
+
+import { EntityId } from "@dust/world/src/EntityId.sol";
+import { Player } from "@dust/world/src/codegen/tables/Player.sol";
+
+import { IAttachProgramHook, IDetachProgramHook } from "@dust/world/src/ProgramInterfaces.sol";
+
+import { AllowedCaller } from "../codegen/tables/AllowedCaller.sol";
+import { Owner } from "../codegen/tables/Owner.sol";
+
+import { SmartItem } from "../codegen/tables/SmartItem.sol";
+import { UniqueEntity } from "../codegen/tables/UniqueEntity.sol";
+
+contract DefaultProgram is IAttachProgramHook, IDetachProgramHook, WorldConsumer {
+  constructor(IBaseWorld _world) WorldConsumer(_world) { }
+
+  function onAttachProgram(EntityId caller, EntityId target, bytes memory extraData) external onlyWorld {
+    uint256 uniqueEntity = UniqueEntity.get() + 1;
+    UniqueEntity.set(uniqueEntity);
+    bytes32 smartItemId = bytes32(uniqueEntity);
+    SmartItem.set(target, smartItemId);
+
+    Owner.set(smartItemId, caller);
+    AllowedCaller.set(smartItemId, caller, true);
```

Comment:
why do we need to create a new `smartItemId` here instead of just using the `target` entity id?

Reply:
See thread above: https://github.com/dustproject/dust/pull/184#discussion_r2032028924

But tl;dr: 
1) User A places down a chest and get entity A
2) Program adds User A to allowed caller list on attach and uses entity A as the key
3) User B mines the chest (detach is called but has a fixed amount of gas, so cleanup logic could fail)
4) User B places down a chest in the same spot and gets entity A (since in the physics entities are tied to positions)
5) **Error**: Now the default program will still incorrectly have User A as an allowed caller

We fix this by creating a new smartItemId, so 5) can't happen